test/action_view/component_test.rb

```diff
@@ -232,6 +232,18 @@ def test_to_component_class
     assert_html_matches "<span>The Awesome post component!</span>", render_inline(post).to_html
   end
 
+  def test_no_initializer
+    if RUBY_VERSION < "2.7"
+      exception = assert_raises NotImplementedError do
+        render_inline(NoInitializerComponent)
+      end
+
+      assert_includes exception.message, "must implement #initialize"
+    else
+      assert_html_matches "Hello, world!", render_inline(NoInitializerComponent).text
+    end
+  end
```

Comment:
It's pretty awkward to have  a test that splits what it asserts based on some logic (the name doesn't line up for example). This is a place where mini_test is weak vs rspec or the like. I wonder can we test the ruby version and then define a method or not? That would allow us to keep the invalid test in the invalid file and selectively include a test here for the happy path case.

Reply:
Good point. I’m not too familiar with MiniTest, but I just found [this](https://www.rubydoc.info/gems/minitest/2.7.0/MiniTest/Assertions#skip-instance_method). So we could probably split it and have a guard clause on each of the respective tests. Something like `skip unless RUBY_VERSION < “2.7”`.

Reply:
I wonder if you can do this:

```
if RUBY_VERSION < '2.7'
  define_method :test_no_initializer do
    ...
  end
end
```

Reply:
Having the invalid components in app/component will also make this a little bit nicer.

Reply:
Do you have thoughts on skip vs. what you recommended? I'm leaning towards `skip` because I see it in the Rails tests. But there's probably instances of the `if … define_method :test_…` way too. 

More context in this comment: https://github.com/github/actionview-component/pull/174#issuecomment-572769270

Reply:
I like skip